packages/block/src/helpers.ts

```diff
@@ -75,3 +79,67 @@ export function getDifficulty(headerData: HeaderData): bigint | null {
   }
   return null
 }
+
+/**
+ * Calculates the excess data gas for a post EIP 4844 block given the parent block header.
+ * @param parent header for the parent block
+ * @param newBlobs number of blobs contained in block
+ * @returns the excess data gas for the prospective next block
+ *
+ * Note: This function expects that it is only being called on a valid block as it does not have
+ * access to the "current" block's common instance to verify if 4844 is active or not.
+ */
+export const calcExcessDataGas = (parent: BlockHeader, newBlobs: number) => {
+  if (!parent._common.isActivatedEIP(4844)) {
+    // If 4844 isn't active on header, assume this is the first post-fork block so excess data gas is 0
+    return BigInt(0)
+  }
+  if (parent.excessDataGas === undefined) {
+    // Given 4844 is active on parent block, we expect it to have an excessDataGas field
+    throw new Error('parent header does not contain excessDataGas field')
+  }
+
+  const consumedDataGas = BigInt(newBlobs) * parent._common.param('gasConfig', 'dataGasPerBlob')
+  const targetDataGasPerBlock = parent._common.param('gasConfig', 'targetDataGasPerBlock')
+
+  if (parent.excessDataGas + consumedDataGas < targetDataGasPerBlock) return BigInt(0)
```

Comment:
Very general EIP question: do you know the reasoning for just capping this at 0 for values below the target? 🤔 One could just have saved the negative value in some form as well?

Reply:
Because the targeting rule (the`fakeExponential` helper) uses the `excessDataGas` in determining the data gas price for the current block.  If that `excessDataGas` value was negative, that would make the data gas price negative which would imply the protocol is paying the transaction submitter to post data gas.  The lowest the `dataGasPrice` can be is 1.

Reply:
Ah thanks a lot for the explanation, that's helpful for some context.